applications/asset_tracker_v2/src/modules/modem_module.c

```diff
@@ -177,6 +183,43 @@ void nrf_modem_fault_handler(struct nrf_modem_fault_info *fault_info)
 	SEND_ERROR(modem, MODEM_EVT_ERROR, -EFAULT);
 }
 
+/* Function that checks the return code from nRF modem library initialization to ensure that
+ * the modem is rebooted if a modem firmware update is ready to be applied or
+ * an error condition occurred during firmware update or library initialization.
+ */
+static void on_modem_lib_init(int ret, void *ctx)
+{
+	ARG_UNUSED(ctx);
+
```

Comment:
```suggestion
if (IS_ENABLED(CONFIG_NRF_CLOUD_FOTA) {
	return;
}
```

Reply:
the modem fota update will be processed when nrf_cloud_init is called. 
i'm not sure how "clean" the reboot will be in ATv2 with this suggested change.

if the reboot/shutdown is messy, it might be better to return to the prior implementation of: 
```
if (IS_ENABLED(CONFIG_NRF_CLOUD_FOTA) {
	(void)nrf_cloud_fota_pending_job_validate(NULL);
}
```
just before the reboot.